mlflow/models/docker_utils.py

```diff
@@ -122,34 +125,44 @@ def _get_mlflow_install_step(dockerfile_context_dir, mlflow_home):
     Get docker build commands for installing MLflow given a Docker context dir and optional source
     directory
     """
-    maven_proxy = _get_maven_proxy()
     if mlflow_home:
         mlflow_dir = _copy_project(src_path=mlflow_home, dst_path=dockerfile_context_dir)
         return (
             f"COPY {mlflow_dir} /opt/mlflow\n"
             "RUN pip install /opt/mlflow\n"
-            "RUN cd /opt/mlflow/mlflow/java/scoring && "
-            f"mvn --batch-mode package -DskipTests {maven_proxy} && "
-            "mkdir -p /opt/java/jars && "
-            "mv /opt/mlflow/mlflow/java/scoring/target/"
-            "mlflow-scoring-*-with-dependencies.jar /opt/java/jars\n"
+            "RUN cd /opt/mlflow/mlflow/java/scoring \\\n && "
+            "SETTINGS_XML=$(mktemp) \\\n && "
+            "python -c 'import mlflow.models.docker_utils as U; print(U.get_maven_settings())'"
```

Comment:
Why not directly get maven setting result in `_get_mlflow_install_step` function ?

Reply:
A couple of reasons, which I outline above:
1. If the proxy has credentials, you can't put the Dockerfile in git without leaking credentials.
2. If the organization that generates the dockerfile has a proxy and it is used by an organization without (or vice-versa), they have to regenerate it. This limits collaboration and sharing of containerized models.
3. If the proxy credentials change (such as having different proxy credentials for each user, or a rotating password), the dockerfile needs to be updated or regenerated when that happens.

The basic idea is that in an ideal world, the same proxy variables set by the `--build-arg`s or the docker `config.json` proxy settings that that are used by pip should also be used by maven, and this accomplishes exactly that.